flink-core/src/main/java/org/apache/flink/core/plugin/PluginLoader.java

```diff
@@ -182,5 +184,20 @@ private boolean isAllowedFlinkClass(final String name) {
 		private boolean isAllowedFlinkResource(final String name) {
 			return Arrays.stream(allowedResourcePrefixes).anyMatch(name::startsWith);
 		}
+
+		static {
+			ClassLoader platformLoader = null;
+			try {
+				platformLoader = (ClassLoader) ClassLoader.class
+					.getMethod("getPlatformClassLoader")
+					.invoke(null);
+			} catch (NoSuchMethodException e) {
```

Comment:
@pnowojski What do you think of the exception handling now? Do you have any reliable way to detect if we are on Java9+?

Reply:
Maybe this?
https://www.journaldev.com/20930/check-java-version#can-we-check-java-version-programmatically
but I'm also kind of fine with as it is.

Reply:
There's no good way to check for java versions on Java 8; Runtime.Version was added in Java 9.